### ISSUE TITLE
This document requires 'TrustedHTML' assignment

### ISSUE THREAD

**Issue by kstubs** [josteink/gmailjs-node-boilerplate#38]
I'm following the ReadMe installation instructions for installing the boilerplate.  I've ran the npm install and build steps and have loaded the extension into chrome.  I've attached my console log.
[mail.google.com-1721145048368.log](https://github.com/user-attachments/files/16253300/mail.google.com-1721145048368.log)

In this log, I see errors related to TrustedHTML and see errors related to jQuery is not defined.  There are more errors, but these might be a cascading result stemming from the above.  I'm not sure.

Is the boilerplate outfitted with the jQuery updates that fix this TrustedHTML issue?

Karl..

**Comment by josteink** [comment]
Good catch! That TrustedHTML patch was never added to the boilerplate repo. 

It would probably be very useful and instructive if added!

Do you think you could be able to create a PR for that?

I'm kinda busy with family vacation atm and while I can quickly approve a PR, implementing it myself might take a bit more time 😇

**Comment by kstubs** [comment]
I will try.  This level of JS, Build, Distribution, etc.. is new to me.  If
I have sometime tonight I will look.  Enjoy your vacay!

**Comment by kstubs** [comment]
Can you point me to the particular pull request  from the gmailjs repo where the TrustedHTML updates were implemented?  This would help get me started in updating the boilerplate to remedy the issues with it.

**Comment by kstubs** [comment]
It looks like the boilerplate is attempting to solved the trusted html policy issue, but I guess not working correctly?

https://github.com/josteink/gmailjs-node-boilerplate/blob/fbb9b306b5debca47270d091f29743f56cc011c3/src/gmailJsLoader.js#L16

**Comment by kstubs** [comment]
I see where you fixed the boiler plate: https://github.com/KartikTalwar/gmail.js/issues/779#issuecomment-1999818556
I'm playing catch-up here!  So I'm not sure what we are trying to fix, since you fixed and tested it and it was working ;/

**Comment by josteink** [comment]
Oh my bad then.

Well, the good news is you got things working, I guess?

**Comment by kstubs** [comment]
Yes/No.  Boilerplate working, yes!  Now working on my custom
implementation.  Unfortunately over the past few years I've lost my
original Gmail code that helped me manage and reply to emails in my Inbox.
As I'm replying to you, I'm thinking through some of the features I had
(this is a thought exercise):


   1. From a popup - Implemented Reply button click
   2. Inject custom template into compose window
   3. Update Subject of email
   4. By changing the subject and including myself in the reply, this
   *custom* subject can then trigger a filter which can then run any of the
   filter routines that you have defined.  For instance, applying a Label to
   the email.
      1. Applying a label to the email was a challenge I had posed to the
      Gmail group in the past.  I don't believe we ever figured that out.
   5. I was also looking up customer details by calling my Api in the
   background script by Ajax calls

FYI - was in Oslo this past May.  I should have looked you up!

Karl Stubsjoen
(602) 845-0006

**Comment by josteink** [comment]
If your extension was published anywhere, it should be easy to just redownload the ZIP/XPI and extract the old code, right? ;)

**Comment by kstubs** [comment]
Hey Kmax,

Unfortunately, I haven't been able to locate that thumb drive.

Regarding the boilerplate issue and the TrustedHTML concern, I believe
those were resolved. The problem I encountered seems to have been caused by
a different extension, not the boilerplate itself.

Specifically, which custom extension, or which features were you interested
in?

Best,
Karl

Karl Stubsjoen
(602) 845-0006

**Comment by krylor** [comment]
Oh, I was looking for a good example where I could inject my custom pop-up as a button.
The idea is to integrate this with my own CRM and respond to events, I had built one based on the earlier boiler-plate example but that's now obsolete and has stopped working altogether.

**Comment by josteink** [comment]
> I had built one based on the earlier boiler-plate example but that's now obsolete and has stopped working altogether.

To be clear... the boilerplate works fine for me. When you say "stopped working", are you talking about the boiler-plate or the project you yourself built?

**Comment by krylor** [comment]
The project that I built based on the boiler-plate broke because Gmail changed a bunch of stuff.
I guess the old boiler-plate is broken too.

**Comment by krylor** [comment]
Can I inject jquery, my legacy requires it.

**Comment by josteink** [comment]
I'm including jquery, but bundling it. 

Once loaded, it should be easy to inject into the global DOM if you need it.
